### PR TITLE
[filesystem] added a method to convert URIs to platform specific paths

### LINKED ISSUE #3709: [filesystem] consider adding a method to convert URIs into path strings
Usually, we pass along platform independent URIs to refer to files. This allows having different OSs for frontend and backend. It requires the path to the script to be encoded with windows file separators `\`
instead of `/`.

I want to run a script from the workspace on the backend. Therefore, I need its platform specific path in order to pass it to the shell (CMD.exe ins this case, as the backend is running on windows). I wanted to use the task API for that, which would require to do the URI->path conversion on the client. But the client has no means to do that. It doesn't actually know what OS the backend runs.

It would be nice if the filesystem exposed a service to the client to do that transformation on the backend.

### PR DESCRIPTION
**Pull request by JanKoehnlein**
Fixes #3709

### CODE REVIEW

**Comment by kittaakos** [comment]
I am sure you have to adjust the mock filesystem...

**Comment by kittaakos** [comment]
As I [said in for the issue](
https://github.com/theia-ide/theia/issues/3709#issuecomment-443660196), I am not a big fan of exposing this as an API to the clients. I think this should be handled by the backend part of your service and not the frontend code but I won't hold it back. I am reviwing it...

File: packages/filesystem/src/node/node-filesystem.ts [hunk 1]
```diff
@@ -379,6 +379,14 @@ export class FileSystemNode implements FileSystem {
         }
     }
 
+    async getFsPath(uri: string): Promise<string | undefined> {
+        if (!uri.startsWith('file:/')) {
+            return undefined;
```
Comment (thegecko) [inline packages/filesystem/src/node/node-filesystem.ts]: @JanKoehnlein Is there any reason this function ignore paths which don't begin with `file:/`? This function would be useful for correctly resolving debug file paths on Windows, however they are of the form `/c:/Users/Rob/Projects/theia-workspace/test.js`

Comment (akosyakov) [reply packages/filesystem/src/node/node-filesystem.ts]: @thegecko Could we wrap a path in URI in the debug extension? We have a contract that we pass normalized URIs between frontend and backend.

Comment (JanKoehnlein) [reply packages/filesystem/src/node/node-filesystem.ts]: @thegecko I find it a bit dangerous to pass non-normalized URIs. From what I understood, we're free to implement all kinds of URI schemes to retrieve / store resources, not necessarily in the file system. The assumption that URIs without a scheme are file URIs should be made in the clients of this API.

Comment (thegecko) [reply packages/filesystem/src/node/node-filesystem.ts]: Acknowledged

@akosyakov 
> Could we wrap a path in URI in the debug extension?

At one point I did implement it this way so this function accepts a URI or string. Happy to go back to that model and keep the `file` check in only when a string is passed in.

**Pull request merged**